xuxuezhou

xuxuezhou opened the pull request:
# JarvisPro ⚡

> **"Technology is best when it brings people together."** – Matt Mullenweg ([source](https://en.wikipedia.org/wiki/Matt_Mullenweg))

**JarvisPro** is your smart task assistant, designed to boost your productivity and streamline your tasks. It is:

- **AI-powered** 🤖  
- **Efficient & User-friendly** 🚀  
- **Super Fast** ⚡  

## How to Use:
1. [Download JarvisPro](#) 📥  
2. Install and launch 🖥️  
3. Add your tasks ✅  
4. Let **JarvisPro** manage everything for you 🤩  

And it's **FREE**! 🎉  

## 📌 Features:
- [ ] Task management  
- [ ] Voice recognition (coming soon)  
- [ ] Automated reminders (coming soon)  

---

## For Java Developers 🎯

If you're a Java developer, you can use **JarvisPro** to practice coding. Here’s an example of the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(JarvisApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

xuxuezhou committed:
Completed Level 0: Rename, Greet, Exit

xuxuezhou committed:
Completed Level 1: Echo feature implemented

xuxuezhou committed:
Completed Level 2: Added task storage and listing functionality

xuxuezhou committed:
Fixed Task class import issue and completed Level 3: Mark as Done

xuxuezhou committed:
Completed Level 4: Integrated ToDos, Deadlines, Events, and Marking functionality

xuxuezhou committed:
Add delete feature with Collection and Enum support

xuxuezhou committed:
Implement Level 7: Task persistence with file storage

xuxuezhou committed:
Merge Level 7: Task persistence with file storage

xuxuezhou committed:
feat: Implement date and time parsing for deadlines and events

xuxuezhou committed:
merge: Integrate Level 8 with date-time parsing

xuxuezhou committed:
Refactor code for better OOP structure and package organization

xuxuezhou committed:
Merge feature-refactor into master with structured OOP and package improvements

xuxuezhou committed:
update .gitignore

xuxuezhou committed:
Merge add-gradle-support into A-Gradle

xuxuezhou committed:
Merge A-Gradle into master

xuxuezhou committed:
Add JUnit tests for Parser and TaskList

xuxuezhou committed:
Merge JUnit test improvements into master

xuxuezhou committed:
Add Shadow JAR support for JAR packaging

xuxuezhou committed:
Merge A-Jar: Add JAR packaging support

xuxuezhou committed:
Add JavaDoc comments for Jarvis class

xuxuezhou committed:
Ensure compliance with Java Coding Standard

xuxuezhou committed:
Add FindCommand to search tasks by keyword

xuxuezhou committed:
Merge A-CodingStandard into master

xuxuezhou committed:
Resolved merge conflicts and merged A-JavaDoc into master

xuxuezhou committed:
Resolved merge conflicts and merged level-9 into master

xuxuezhou committed:
Fix: resolve text UI testing issue

xuxuezhou committed:
Create extension to handle error

kylieluk88 commented on src/main/java/jarvis/util/Parser.java:
```diff
@@ -0,0 +1,86 @@
+package jarvis.util;
+
+import jarvis.command.*;
```
Should list imports explicitly.

kylieluk88 commented on src/main/java/jarvis/util/Storage.java:
```diff
@@ -0,0 +1,85 @@
+import jarvis.task.ToDo;
+import jarvis.task.Deadline;
+import jarvis.task.Event;
+import java.io.*;
```
Should list imports explicitly

kylieluk88 reviewed (commented):
Overall, style is clear and does not violate coding standards. Good job!

kylieluk88 commented on src/main/java/jarvis/util/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+     * @return The list of tasks.
+     */
+    public List<Task> getTasks() {
+        return new ArrayList<>(tasks); // 返回任务列表的副本，防止外部修改内部数据
```
try to keep all your comments in English so that others reviewing the code can understand :)

xuxuezhou committed:
Use checkStyle to detect coding style violations, with bugs

xuxuezhou committed:
Implement A-CheckStyle changes

xuxuezhou committed:
Merge A-CheckStyle into master

xuxuezhou committed:
Add GUI for Jarvis

xuxuezhou committed:
Imcrement JavaDoc

jensenhuangyankai commented on src/main/java/jarvis/Jarvis.java:
```diff
@@ -0,0 +1,75 @@
+        tasks = new TaskList();
+    }
+
+    public String getResponse(String input) {
```
Good to include DocStrings for every function

jensenhuangyankai commented on src/main/java/jarvis/task/Task.java:
```diff
@@ -0,0 +1,77 @@
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
```
Good job following camelCase scheme for functions

xuxuezhou committed:
Fix Jarvis exit behavior and clear chat on 'bye' command

Previously, the 'bye' command did not properly clear the chat window or close the JavaFX application. This commit ensures that when 'bye' is entered, Jarvis will clear all messages from the UI and close the application window.

Changes include:
-
  - Added  to remove all chat messages.
  - Added  and  to manage the JavaFX primary stage.
  - Ensured  is properly assigned in .

-
  - Modified  to detect the 'bye' command.
  - Calls  before closing the window.

-
  - Ensures  receives a reference to , allowing it to properly close the application when needed.

With these changes, Jarvis now gracefully exits and resets the chat history when the user types 'bye'.

xuxuezhou committed:
Merge tag 'branch-Level-10'

xuxuezhou committed:
[A-Assertions] Add assertions to ensure validity of key objects and inputs

- Added assertions to check that UI, Storage, and TaskList are properly initialized.
- Ensured that user input is not null before processing.
- Verified that parsed commands are not null before execution.
- Checked that the JavaFX stage is not null before closing the application.

These assertions help catch potential issues early during debugging and ensure that critical assumptions hold in the program.

xuxuezhou committed:
Refactor command classes to improve code quality

- Extracted static constants for messages to avoid magic strings.

- Added  method to simplify index validation logic.

- Reduced deep nesting by using early return.

- Ensured  method in  is used instead of direct list access.

- Improved readability and maintainability of , , and .

Tag: A-CodeQuality

xuxuezhou committed:
feat: Refactor AddCommand to use Java Streams

- Replaced manual string concatenation with  for improved readability and efficiency.
- Introduced  constant to remove magic strings.
- Enhanced the  method to provide a formatted list of tasks using Streams.

This update aligns with the A-Streams requirement by leveraging Java Streams for data processing.

xuxuezhou committed:
Merge tag 'A-Streams'

xuxuezhou committed:
Add extension to detect duplicates

xuxuezhou committed:
Improve feature with auto-expanding and highlight error message

xuxuezhou committed:
Update README.md

xuxuezhou committed:
Update conversion content

xuxuezhou committed:
Remove writing file issue and update README

xuxuezhou committed:
Update Ui iamge

xuxuezhou committed:
Merge with branch-A-Assertions

xuxuezhou committed:
Remove GPT feature

xuxuezhou committed:
Add FindCommand and Fix formatting bug and Remove dead code

xuxuezhou committed:
Update with new Jar file
